components/navigable-container/README.md

```diff
@@ -0,0 +1,156 @@
+
+- Type: `String`
+- Required: No
+
+### onNavigate
+
+A callback invoked when the menu navigates to one of its children passing the index and child as an argument
+
+- Type: `Function`
+- Required: No
+
+### deep
+
+A boolean to look for navigable children in the direct children or any descendant. True means that any descendant can be considered navigable, and false means only direct children are considered.
+
+- Type: `Boolean`
+- Required: No
+- default: false
+
+
+### widget
+
+A boolean which specifies whether or not the focusable elements in this component might contain the focus, rather than have the focus. This is often used for `TabbableContainer` layouts that might have arrow key navigation inside each tabstop.
+
+- Type: `Boolean`
+- Required: No
+- default: true for `TabbableContainer`, false otherwise.
+
+
+## NavigableMenu
+
+
+A NavigableMenu allows movement up and down (or left and right) the component via the arrow keys. The `tab` key is not handled. The `orientation` prop is used to determine whether the arrow keys used are vertical or horizontal.
+
+### Usage
+
+```jsx
+import { NavigableMenu, Button } from '@wordpress/components';
+
+function MyMenu() {
+	return (
+		<NavigableMenu>
+			<Button>My Button 1</Button>
+			<Button>My Button 2</Button>
+			<Button>My Button 3</Button>
+		</NavigableMenu>
+	);
+}
+```
+
+### Additional Props
+
+#### orientation
+
+The orientation of the menu. It could be "vertical" or "horizontal"
+
+- Type: `String`
+- Required: No
+- Default: `"vertical"`
+
+#### stopArrowKeys
+
+A boolean which specifies whether to stop the arrow keys that are not being used for navigation from bubbling up the DOM tree. This can be useful if you have a horizontal menu, but you don't want up and down to move the page.
+
+- Type: `Boolean`
+- Required: No
+- Default: true
+
+
+#### cycle
+
+A boolean which tells the component whether or not to cycle from the end back to the beginning and vice versa. If it is false, then as soon as navigation moves outside the component, the keydown event will not be handled.
+
+- Type: `Boolean`
+- Required: No
+- default: true
+
+----
+
+## NavigableGrid
+
+A NavigableGrid allows movement up and down and left and right through a grid of components using the arrow keys. Note, it expects that the grid is laid out in Left to Right, Top to Bottom DOM ordering. Navigable grids will always cycle.
+
+### Usage
+
+```jsx
+import { NavigableGrid, Button } from '@wordpress/components';
+
+function MyGrid() {
+	return (
+		<NavigableGrid>
+			<Button>My Button 1</Button>
+			<Button>My Button 2</Button>
+			<Button>My Button 3</Button>
+      <Button>My Button 4</Button>
+		</NavigableGrid>
+	);
+}
+```
+### Additional Props
+
+### width
+
+An integer specifying the number of components that are displayed across the screen in one row. This relates to how the arrow keys will determine what to focus next.
```

Comment:
My personal preference would be to drop this in favor of having the `right` behave exactly as `down`...
What if we have incomplete rows, this `width` prop won't be sufficient to ensure the right behavior

Reply:
This particular mode is designed to only handle a situation where the grid is laid out visually in rows (due to wrapping) but there no DOM structure that enforces those rows (so it's essentially just CSS). I found that when playing with the InsertMenu in the blocks section, pressing `down` and seeing the focus shift to the right was very disconcerting, so this is just to allow a nice clean navigation for grids with CSS wrapping. The problem is that you either need to calculate the width at run-time (via y position changes), or know it in advance (which you might know depending on how your CSS is making it show in different rows).

This code is designed to work for that situation for a regular grid. The regular grid can have fewer columns on its last row, but the last row is the only one that can be different. It is basically an exact match for what I was seeing with the block categories on the Inserter Menu. We've used it in the past for emoticon panels etc. Users tend to expect to be able to navigate across a grid with the arrow keys. 

Having said all that, if you do want to avoid doing this, and only use grid navigation for situations when each row is a new DOM structure (like a table), then that makes sense. It means there is a lot less guesswork, obviously.

Reply:
We had this kind of navigation in the firsts iterations of the Inserter, but for some reason, it was reverted to a simpler alternative. Shame that I don't recall the reason and I couldn't find the PR responsible for that neither :(

Reply:
I've reverted it to just use down to the move to the right.